### DISCUSSION TITLE
Control of color?

### DISCUSSION THREAD

**Discussion by jhdedrick** [marcusolsson/grafana-treemap-panel#11] [Ideas]
Thanks to the devs for this plugin!

Is there a way to control the color independently of the size of the boxes?  This would add a degree of freedom and really increase the amount of information you could communicate.

**Comment by marcusolsson** [comment]
Thanks for trying it out!

The treemap is using the color scheme set in the field options. 

You can either set it to use a gradient based on the value:

<img width="1435" alt="CleanShot 2021-02-11 at 20 31 39@2x" src="https://user-images.githubusercontent.com/8396880/107688668-2e2def80-6ca8-11eb-8104-f622dbd3ad2c.png">

Or, use thresholds to color by ranges:

<img width="1012" alt="CleanShot 2021-02-11 at 20 33 09@2x" src="https://user-images.githubusercontent.com/8396880/107688934-806f1080-6ca8-11eb-859a-128558c5083a.png">

**Reply by jhdedrick** [reply]
Thanks Marcus.   But since the value is already indicated visually by area, it would be great if the color could be controlled separately, so it can be used to display something else.  For example, I'm considering using treemap to display the capacity consumption of (a large number of) users of a cloud storage system.  The area conveniently indicates which are the big consumers.  But I'd like to use  color to indicate what fraction of their quota each user is consuming - something that is totally independent of their actual consumption.

**Reply by marcusolsson** [reply]
Oh, I see! So you'd like to be able to use a separate dimension for the color basically? I could add a **Color by** dimension that defaults to the same field as used by Size, but that could also be set separately. Would that work for you?

**Reply by jhdedrick** [reply]
Perfect!

**Reply by marcusolsson** [reply]
Added with https://github.com/marcusolsson/grafana-treemap-panel/pull/12
